pymbolic/primitives.py

```diff
@@ -1796,13 +1796,13 @@ def wrap_in_cse(expr, prefix=None):
         if prefix is None:
             return expr
         if expr.prefix is None and type(expr) is CommonSubexpression:
-            return CommonSubexpression(expr.child, prefix)
+            return CommonSubexpression(expr.child, prefix, cse_scope.EVALUATION)
 
         # existing prefix wins
         return expr
 
     else:
-        return CommonSubexpression(expr, prefix)
+        return CommonSubexpression(expr, prefix, cse_scope.EVALUATION)
```

Comment:
This function should just have a `scope` argument, with a default.

Reply:
Added one in https://github.com/inducer/pymbolic/compare/9a62b35b5063d40cdbbb2084414e124141f3406b..7eba5378d83bfd5d6146ed9a94f16478758a801a.